server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -241,7 +248,12 @@ public DocumentFieldMappers mappers() {
     }
 
     public ParsedDocument parse(SourceToParse source) throws MapperParsingException {
-        return documentParser.parseDocument(source);
+        return documentParser.parseDocument(source, mapping.metadataMappers);
+    }
+
+    public ParsedDocument createTombstoneDoc(String index, String type, String id) throws MapperParsingException {
+        final SourceToParse emptySource = SourceToParse.source(index, type, id, new BytesArray("{}"), XContentType.JSON);
```

Comment:
do we need some identifier that this doc is a tombstone? I am not sure we do at this point but down the road we would no?

Reply:
+1 .  We don't plan to keep many of these around so the overhead is minimal and we'd have the ability to search for them and debug. The question is how to do this. I guess the easiest would be to add a boolean metadata field to  the mapping, but that feels like an overkill. In any case, I think this can be a follow up.

Reply:
Yes, I will think about it and make it in a follow-up.